src/lib/warn-upload-overwrite.js

```diff
@@ -15,139 +16,195 @@ const responseChoicesWithoutDiff = [
 ];
 const responseChoicesWithDiff = responseChoicesWithoutDiff.concat([ 'View diff' ]);
 
-const getRevsAndHashesDocKey = () => {
+const getEnvironmentKey = () => {
   const parsed = url.parse(environment.apiUrl);
   const key = `${parsed.hostname}${parsed.pathname || 'medic'}`;
   return key;
 };
 
-const preUploadByRev = async (db, doc) => {
-  let localDoc = JSON.parse(JSON.stringify(doc));
+const getBookmarksDir = () => path.join(environment.pathToProject, '.bookmarks');
```

Comment:
Interesting naming "bookmarks". Is this named with the intention of storing more things in the future? 
I would have called in "snapshots". 
No need to change, I just got strangely fixated onto this.

Reply:
When I first made this change it was storing the hashes and revs - now everything's a hash. I like snapshots - I'll make the change.

Reply:
Done.